compiler/rustc_monomorphize/src/errors.rs

```diff
@@ -0,0 +1,74 @@
+use std::path::PathBuf;
+
+use rustc_errors::ErrorGuaranteed;
+use rustc_macros::SessionDiagnostic;
+use rustc_session::SessionDiagnostic;
+use rustc_span::Span;
+
+#[derive(SessionDiagnostic)]
+// FIXME: This should be #[fatal...], but that doesn't exist, yet...
+#[error(monomorphize::recursion_limit)]
+pub struct RecursionLimit {
+    #[primary_span]
+    pub span: Span,
+    pub shrunk: String,
+    #[note]
+    pub def_span: Span,
+    pub def_path_str: String,
+    #[note(monomorphize::written_to_path)]
+    pub was_written: Option<()>,
```

Comment:
Does this work if you do `Option<PathBuf>` and then use `was_written` only in the `note` attached to this field? I *think* that might work.

Reply:
I tried that first! Unfortunately, it gives an error that `#[note]` can only be applied to `Span` or `()`. 😞